### PR TITLE
BaSP-MR-108: Implementation redux in admins

### PR DESCRIPTION
**Pull request by Eugeniagassman**
implement redux in admin, creating the thunks, reducer, constants, and actions files.
 in this implementation the get, put, post, and delete methods are executed.

### CODE REVIEW

File: src/Components/Admins/Form/index.js [hunk 1]
```diff
@@ -27,61 +29,40 @@ const AdminForm = () => {
...
+  useEffect(() => {
+    formEdit(id);
+  }, []);
 
-      const createdAdminsData = await createdAdmin.json();
-      if (createdAdmin.ok) {
+  const formEdit = (id) => {
+    if (id) {
+      const data = admins.find((aux) => aux._id === id);
+      console.log('data', data);
```
Comment (Tilajc) [inline src/Components/Admins/Form/index.js]: you can delete this console.log
```suggestion

```

File: src/Redux/Admins/actions.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  };
+};
+
+export const putAdminsLoading = () => {
+  return {
+    type: PUT_ADMINS_LOADING
+  };
+};
+
+export const putAdminsSuccess = (updAdmin) => {
```
Comment (Tilajc) [inline src/Redux/Admins/actions.js]: here when you add an id the response will be correct
```suggestion
export const putAdminsSuccess = (updAdmin, id) => {
```

File: src/Redux/Admins/actions.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  return {
+    type: PUT_ADMINS_SUCCESS,
+    payload: updAdmin
```
Comment (Tilajc) [inline src/Redux/Admins/actions.js]: same here, you need the id
```suggestion
    payload: updAdmin, id
```

File: src/Redux/Admins/reducer.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    case DEL_ADMINS_LOADING: {
+      return {
+        ...state,
+        isLoading: true,
+        error: null
+      };
+    }
+    case DEL_ADMINS_SUCCESS: {
+      const idDeletedAdmin = [action.payload.id];
```
Comment (Tilajc) [inline src/Redux/Admins/reducer.js]: the ID without _ is undefined
```suggestion
      const idDeletedAdmin = [action.payload._id];
```

File: src/Redux/Admins/reducer.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      const currentAdmins = state.data.filter((admins) => admins.id !== idDeletedAdmin);
```
Comment (Tilajc) [inline src/Redux/Admins/reducer.js]: same here
```suggestion
      const currentAdmins = state.data.filter((admins) => admins._id !== idDeletedAdmin);
```

**Review by Tilajc — CHANGES_REQUESTED** [review]
well done! just a little changes

**Comment by Tilajc** [comment]
I don't know how to made a comment when you didn't make any change there, but in the form index, into the html, you need to call the spinner too

**Comment by Eugeniagassman** [comment]
[Implement Redux on the Admins entity](https://trello.com/c/yQkwiK9d/108-implement-redux-on-the-admins-entity)

**Review by Tilajc — APPROVED** [review]
Good job!

**Comment by Flor-Farruggia** [comment]
Nice job!

**Pull request merged**